### PR TITLE
remove generated base image sources

### LINKED ISSUE #488: remove base-image sources from kind binary
**What should be cleaned up or changed**: we should remove the code generation / baked in sources for `kind build base-image`.

**Why is this needed**: These are unnecessary bloat in retrospect, if you wish to build the base image you're going to want to be able to tweak the sources, we did not wind up exposing any other configuration options for this build related to the contents.

We can remove this, and instead by default autodetect the kind sources location the same as we do for kubernetes, we already have an explicit option to select the sources.

/assign

cc @aojea @neolit123 @tao12345666333 @fabriziopandini

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes #488 
EDIT: also #485, we can't find a good lower bound for docker for desktop because it appears to be lower than the settings offer, but we can stop suggesting that it is 4 core / 4 cpu unless you are building kubernetes.

### CODE REVIEW

File: pkg/build/base/base.go [hunk 1]
```diff
@@ -79,22 +80,20 @@ func (c *BuildContext) Build() (err error) {
...
-	// if SourceDir is unset, use the baked in sources
+	// if SourceDir is unset then try to autodetect source dir
 	buildDir := tmpDir
 	if c.sourceDir == "" {
-		// populate with image sources
-		err = sources.RestoreAssets(buildDir, "images/base")
+		pkg, err := build.Default.Import("sigs.k8s.io/kind", build.Default.GOPATH, build.FindOnly)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to locate sources: %v", err)
```
Comment (tao12345666333) [inline pkg/build/base/base.go]: maybe using errors instead of fmt.

Comment (BenTheElder) [reply pkg/build/base/base.go]: done, also updated the docs

**Comment by BenTheElder** [comment]
updated the docs, includes minor updates related to https://github.com/kubernetes-sigs/kind/issues/485 as well as the changes here

**Pull request merged**